sha1_mb/sha1_mb_rand_update_test.c

```diff
@@ -116,15 +117,24 @@ int main(void)
 			ctx = sha1_ctx_mgr_submit(mgr,
 						  &ctxpool[i],
 						  buf_ptr[i], UPDATE_SIZE, HASH_UPDATE);
-
-		// Add jobs while available or finished
-		if ((ctx == NULL) || hash_ctx_complete(ctx)) {
-			i++;
+		if (ctx != NULL && hash_ctx_idle(ctx)) {
```

Comment:
This doesn't seem quite right.  The context status could be idle or complete.  We know we we have a free lane if the previous submit returned null or complete.  I think this is trying to compensate for some other issue.

Reply:
> 
> 
> One comment plus does md5 have similar issue with return value?

md5 have the issue also . I will update it . 
And I will check the context status again .

When I write sha1_mb code , I set the unused_lanes to 0xf0123 then it can not pass rand test . 
At the sha*_mb commits , I set the unused_lanes to right order .
But I think the real world will appear same issue with some submit sequence and the commit can fix the issue . That will give end user a good example for how to use multi-buffer api .

Reply:
@gbtucker  I think the code is right . 
if the previous submit returned null or complete , the loop at line 134 will process those cases .

Reply:
What does the unused lanes = 0xf0123 represent?  Can you obtain this by a sequence of submissions?

Could it be that sometimes your implementation does not run to completion of the shortest hash when the lanes are full?

Reply:
I will reproduce the case and update the test 

"unused_lanes" is member of  SHA*_MB_JOB_MGR , it is used the allocate lanes. 

For the sse implementation , it is initialed at sha1_mb_mgr_init_sse.c and 
sha1_mb_mgr_submit_sse.asm line 117 load and modify the variable .

Reply:
@gbtucker , I post a patch here . It can reproduce the issue I said